src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -60,8 +62,41 @@ def handleRequest(request)
         response["X-Puppet-Version"])
   end
 
-  def compileCatalog(_request_data)
-    {}
+  def convert_java_args_to_ruby(hash)
+    Hash[hash.collect do |key, value|
+        # Stolen and modified from params_to_ruby in handler.rb
+        newkey = key.to_s
+        newkey.slice!(0)
+        if value.java_kind_of?(Java::ClojureLang::PersistentArrayMap)
```

Comment:
Not sure if this`Java::ClojureLang::PersistentArrayMap` should be `Java::JavaUtil::List` like it is in the original method this is taken from.

Reply:
list is handled in the next branch

more importantly, this is not a reliable way to determine if something is a clojure map since the underlying class can change dynamically based on the size of the map

Reply:
Do you know what would be more reliable here?

Reply:
I would assume something like `Java::ClojureLang::IPersistentMap`? (See: [clojure's java interop docs](https://clojure.org/reference/data_structures#Maps) and [the java code](https://github.com/clojure/clojure/blob/4ef4b1ed7a2e8bb0aaaacfb0942729252c2c3091/src/jvm/clojure/lang/IPersistentMap.java), it looks like Java::Iterable would work, though I'd be surprised if that didn't also get Vectors as well).

Reply:
It looks like [APersistentMap](https://github.com/clojure/clojure/blob/master/src/jvm/clojure/lang/APersistentMap.java) would also work?

Reply:
using the interface makes the most sense i think